src/cast_core/src/CommitLatency.cpp

```diff
@@ -0,0 +1,199 @@
+                  // database, or collection. (CXX-1748)
+                  collection.read_concern(rc);
+              }
+          }
+};
+
+void CommitLatency::run() {
+    using namespace bsoncxx::builder::stream;
+
+    for (auto&& config : _loop) {
+
+        std::shared_ptr<mongocxx::client_session> _session;
+        if (config.begin() != config.end()) {
+              BOOST_LOG_TRIVIAL(info) << "Starting " << config->threads << "x" << config->repeat 
+                                      << " CommitLatency transactions (2 finds, 2 updates, 1 aggregate).";
+              BOOST_LOG_TRIVIAL(info) << "CommitLatency options: wc=" 
+                                      << bsoncxx::to_json(config->wc.to_document())
+                                      << " rc="
+                                      << config->rc.acknowledge_string()
+                                      << " rp="
+                                      << config->rp_string
+                                      << " session="
+                                      << (config->useSession ? "true" : "false")
+                                      << " transaction="
+                                      << (config->useTransaction ? "true" : "false");
+
+
+              // start_session requires access to _client object, and session objects are of course
+              // per thread, so we initialize it here.
+              if(config->useSession || config->useTransaction) {
+                  // Sessions have causal consistency by default.
+                  // It's also possible to unset that, but we don't provide a yaml option to actually do that.
+                  _session = std::make_shared<mongocxx::client_session>(_client->start_session());
+              }
+        }
+        for (const auto&& _ : config) {
+            _strategy.run(
+                [&](metrics::OperationContext& ctx) {
+                    // Basically we withdraw `amount` from account 1 and deposit to account 2
+                    // amount = random.randint(-100, 100)
+                    auto amount = config->amountDistribution(_rng);
+
+                    if (config->useTransaction) {
+                        _session->start_transaction(config->optionsTransaction);
+                    }
+
+                    // result1 = db.hltest.find_one( { '_id': 1 }, session=session )
+                    bsoncxx::document::value doc_filter1 = document{}
+                        << "_id" << 1
+                        << finalize;
+                    auto result1 = config->collection.find_one(*_session, doc_filter1.view(), config->optionsFind);
+
+                    // db.hltest.update_one( {'_id': 1}, {'$inc': {'n': -amount}}, session=session )
+                    bsoncxx::document::value doc_update1 = document{}
+                        << "$inc" << open_document
+                            << "n" << -amount
```

Comment:
Please confirm that that these streams are indented correctly after clang format. (You shouldn't need to format anything manually)

Reply:
What is "correctly"? clang-format of course isn't aware of idiomatic mongocxx style and has mangled these to single line. It's not clear to me that's what you expected?